tests/sweep_framework/sweep_utils/max_pool2d_common.py

```diff
@@ -94,6 +94,8 @@ def run_max_pool2d(
         memory_config=memory_config,
         applied_shard_scheme=sharding,
         in_place_halo=in_place,
+        deallocate_input=True,
```

Comment:
since sweeps start from DRAM, with right defaults it should deallocate input after halo.
It's important to have right defaults and not tweak any advance params in sweeps as compilers won't do that.